webapp/src/app.tsx

```diff
@@ -264,6 +266,17 @@ export class ProjectView
         this.shouldTryDecompile = false;
     }
 
+    openSerial(isSim: boolean) {
+        if (pxt.appTarget.serial && pxt.appTarget.serial.useEditor) {
+            let event = "serial." + (isSim ? "simulator" : "device") + "EditorOpened"
+            pxt.tickEvent(event)
+            this.serialEditor.setSim(isSim)
+            this.setFile(pkg.mainEditorPkg().lookupFile("this/" + pxt.SERIAL_EDITOR_FILE))
```

Comment:
This is a little problematic, as the serial file might not exist (if you hadn't created a new project)